php/libraries/Utility.class.inc

```diff
@@ -933,5 +935,100 @@ class Utility
         }
         return $languages;
     }
+
+    /**
+     * Updates the password for this user into the appropriate field in the
+     * database.
+     *
+     * @param \User     $user     The plain text password to be hashed and saved.
+     * @param \Password $password The plain text password to be hashed and saved.
+     * @param ?DateTime $expiry   When the password will expire. Can be set to a
+     *                            date in the past in order to create a one-time
+     *                            password.
+     *
+     * @return void
+     */
+    static function updatePassword(
```

Comment:
I think this belongs to the User class.

Reply:
See PR description and Dave's linked PR for why this was moved from the User class

Reply:
From what I can see, @driusan is looking for a model/data-transfer-object(dto) representation of a User. Our current User class is a controller, to deal with CRUD stuff related to a user as well as a model for a user. By striping all business related logic, what remain should be close to a model. but we need to find a place to put all the logic. Moving it to Utility is not a sustainable solution.

Reply:
I agree that Utility probably isn't the best solution long-term but I'm curious whether the functions should stay in User for now or not.

@driusan do you have an opinion?